aten/src/ATen/native/Linear.cpp

```diff
@@ -47,7 +47,8 @@ Tensor linear(const Tensor& input, const Tensor& weight, const c10::optional<Ten
   auto output = at::matmul(input, weight.t());
   if (bias->defined()) {
     // for composite compliance use out-of-place version of `add`
-    if (isTensorSubclassLike(*bias)) {
+    if (isTensorSubclassLike(*bias) ||
+        (*bias)._fw_grad(/*level*/ 0).defined()) {
```

Comment:
Is this the correct way to check if forward mode AD is being used?

Reply:
Should use https://github.com/pytorch/pytorch/blob/855c4eb0514cd8919b2f91a840c950304bbece66/torch/csrc/autograd/functions/utils.h#L90-L92 as the check

Reply:
nit: `bias->_fw_grad(/*level*/ 0).defined()), but yeah that should be fine!

Reply:
Done.